src/spikeinterface/core/sortingresult.py

```diff
@@ -0,0 +1,1324 @@
+            zarr_root = self._get_zarr_root()
+            zarr_root.create_dataset("random_spikes_indices", data=self.random_spikes_indices)
+
+    def get_selected_indices_in_spike_train(self, unit_id, segment_index):
+        # usefull for Waveforms extractor backwars compatibility
+        # In Waveforms extractor "selected_spikes" was a dict (key: unit_id) of list (segment_index) of indices of spikes in spiketrain
+        assert self.random_spikes_indices is not None, "random spikes selection is not computeds"
+        unit_index = self.sorting.id_to_index(unit_id)
+        spikes = self.sorting.to_spike_vector()
+        spike_indices_in_seg = np.flatnonzero((spikes["segment_index"] == segment_index) & (spikes["unit_index"] == unit_index))
+        common_element, inds_left, inds_right  = np.intersect1d(spike_indices_in_seg, self.random_spikes_indices, return_indices=True)
+        selected_spikes_in_spike_train = inds_left
+        return selected_spikes_in_spike_train
+
+
+global _possible_extensions
+_possible_extensions = []
+
+def register_result_extension(extension_class):
+    """
+    This maintains a list of possible extensions that are available.
+    It depends on the imported submodules (e.g. for postprocessing module).
+
+    For instance with:
+    import spikeinterface as si
+    only one extension will be available
+    but with 
+    import spikeinterface.postprocessing
+    more extensions will be available
+    """
+    assert issubclass(extension_class, ResultExtension)
+    assert extension_class.extension_name is not None, "extension_name must not be None"
+    global _possible_extensions
+
+    already_registered = any(extension_class is ext for ext in _possible_extensions)
+    if not already_registered:
+        assert all(
+            extension_class.extension_name != ext.extension_name for ext in _possible_extensions
+        ), "Extension name already exists"
+
+        _possible_extensions.append(extension_class)
+
+
+def get_extension_class(extension_name: str):
+    """
+    Get extension class from name and check if registered.
+
+    Parameters
+    ----------
+    extension_name: str
+        The extension name.
+
+    Returns
+    -------
+    ext_class:
+        The class of the extension.
+    """
+    global _possible_extensions
+    extensions_dict = {ext.extension_name: ext for ext in _possible_extensions}
+    assert extension_name in extensions_dict, "Extension is not registered, please import related module before"
+    ext_class = extensions_dict[extension_name]
+    return ext_class
+
+
+class ResultExtension:
+    """
+    This the base class to extend the SortingResult.
+    It can handle persistency to disk any computations related
+
+    For instance:
+      * waveforms
+      * principal components
+      * spike amplitudes
+      * quality metrics
+
+    Possible extension can be register on the fly at import time with register_result_extension() mechanism.
```

Comment:
```suggestion
    A possible extension can be registered on-the-fly at import time with the register_result_extension() mechanism.
```
Other parts of the code base use on-the-fly stylized with hyphens. So probably should just add here.